packages/popover/src/index.tsx

```diff
@@ -144,12 +146,17 @@ const usePopover = ({
       ...getPlacement(targetRect, popoverRect)
     };
 
+    if (isFlipped !== shouldFlip) {
```

Comment:
`isFlipped` is not passed as a dependency for `useCallback`.
you should pass it or maybe the approach below might be better for perf. so you do not have to pass `isFlipped` as a dependency. _(please test the change)_
```
setIsFlipped(prev => {
  if (prev !== shouldFlip) return shouldFlip;
  return prev;
})
```

Reply:
You're right; there's a missing dependency, and I tested your solution. 

However, I have another idea for better performance: We can keep the `isFlipped` state as ref with useRef instead of `useState`. We don't have to `useState` because we can't change the isFlipped state directly, this state can change with only a way: the `getAvailiableSpaces` function. So this fn, its dependencies (especially `children`) and also two listeners (`resize`, `scroll`) can control and trigger this state change. 

So I guess `isFlipped` is just a flag. This approach may be better perf and can make us avoid cases that I can't even guess right now.

Btw, I tried all 3 approaches that work correctly and the same re-render perf.

```
 const usePopover = ({
  hidden,
  autoFlip,
  children,
  placement,
  targetRef,
  popoverNode
}: IUsePopoverProps) => {
  const [styles, setStyles] = useState<CSSProperties>(defaultPopoverStyles);
  const isFlippedRef = useRef(false);

  const getAvailableSpaces = useCallback(
    (targetRect: DOMRect, popoverRect: DOMRect): Record<string, boolean> => {
   ...

  const getPopoverPosition = useCallback(() => {
    if (!popoverNode) return;

    const targetRect: DOMRect = targetRef.current.getBoundingClientRect();
    const popoverRect: DOMRect = popoverNode.getBoundingClientRect();

    const availableSpaces = getAvailableSpaces(targetRect, popoverRect);

    const shouldFlip =
      autoFlip &&
      !availableSpaces[placement] &&
      availableSpaces[OppositePlacements[placement]];

    const certainPlacement = shouldFlip
      ? OppositePlacements[placement]
      : placement;

    const getPlacement = getPlacementGetter(certainPlacement);
    const newPosition: CSSProperties = {
      ...getPlacement(targetRect, popoverRect)
    };

    isFlippedRef.current = shouldFlip; 

    setStyles((prev) => ({
      ...prev,
      ...newPosition
    }));
  }, [
    hidden,
    autoFlip,
    children,
    placement,
    popoverNode,
    getAvailableSpaces,
    getPlacementGetter
  ]);

  useIsomorphicLayoutEffect(() => {
    getPopoverPosition();
  }, [getPopoverPosition]);

  useEventListener({
    name: 'resize',
    listener: getPopoverPosition
  });

  useEventListener({
    name: 'scroll',
    options: { passive: true },
    listener: getPopoverPosition
  });

  return {
    styles,
    isFlipped: isFlippedRef.current
  };
};
```

Waiting your reply to move forward. 🏎️